packages/core/src/lib/models/node.model.ts

```diff
@@ -193,17 +193,17 @@ export class NodeModel extends BaseModel<DiagramModel> {
       .pipe(this.withLog('selectHeight'));
   }
 
-  setExtras<E>(extras: Partial<E>) {
-    this.extras$.set(extras);
+  setExtras(extras: Partial<Extras>): void {
```

Comment:
I think it's worth adding an `upsert` method for the extras that will merge the current state with the changes, instead of doing that in the `set` method

Reply:
I agree, but a question came to mind:
currently extras can be of any type, not only objects.
do we want to force it to be an object?

Reply:
then let's avoid spreading, keep the set method as is just with the new generics.
the user can handle the logic of adding on top of the current extras according to the data structure he provided.